Source/ACE/Entity/Lifestone.cs

```diff
@@ -1,63 +1,88 @@
-                                                                 player.Sequences, motionSanctuary);
-
-                // This event was present for a pcap in the training dungeon.. Why? The sound comes with animationEvent...
-                var soundEvent = new GameMessageSound(Guid, Sound.LifestoneOn, 1);
-
-                if (player.Location.SquaredDistanceTo(Location) >= radiusSquared)
+                if (player == null)
                 {
-                    serverMessage = "You wandered too far to attune with the Lifestone!";
+                    return;
                 }
+
+                if (!player.IsWithinUseRadiusOf(this))
+                    player.DoMoveTo(this);
                 else
                 {
-                    player.SetCharacterPosition(PositionType.Sanctuary, player.Location);
+                    ActionChain useObjectChain = new ActionChain();
 
-                    // create the outbound server message
-                    serverMessage = "You have attuned your spirit to this Lifestone. You will resurrect here after you die.";
-                    player.HandleActionMotion(motionSanctuary);
-                    player.Session.Network.EnqueueSend(soundEvent);
-                }
+                    useObjectChain.AddAction(player, () =>
+                    {
+                        Activate(player);
+                    });
 
-                var lifestoneBindMessage = new GameMessageSystemChat(serverMessage, ChatMessageType.Magic);
-                // always send useDone event
-                var sendUseDoneEvent = new GameEventUseDone(player.Session);
-                player.Session.Network.EnqueueSend(lifestoneBindMessage, sendUseDoneEvent);
+                    useObjectChain.AddDelaySeconds(5); // TODO: get animation frames length and put that delay here
+
+                    useObjectChain.AddAction(player, () => player.SetCharacterPosition(PositionType.Sanctuary, player.Location));
+
+                    useObjectChain.AddAction(player, () => player.Session.Network.EnqueueSend(new GameMessageSystemChat("You have attuned your spirit to this Lifestone. You will resurrect here after you die.", ChatMessageType.Magic)));
+
+                    var sendUseDoneEvent = new GameEventUseDone(player.Session);
+                    useObjectChain.AddAction(player, () => player.Session.Network.EnqueueSend(sendUseDoneEvent));
+
+                    useObjectChain.EnqueueChain();
+                }
             });
             chain.EnqueueChain();
         }
+
+        private void Activate(Player activator)
+        {
+            CurrentLandblock.EnqueueBroadcastMotion(activator, motionSanctuary);
+
+            CurrentLandblock.EnqueueBroadcastSound(this, Sound.LifestoneOn); // This event was present for a pcap in the training dungeon.. Why? The sound already comes with animationEvent...
+
+            if (activator.Guid.Full > 0)
+                UseTimestamp++;
```

Comment:
If your not doing an increment, why would you make the code look like your doing an increment?  This is poorly designed and makes no sense.